lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,25 +226,65 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
+    }
+    #The transaction commit and temporary allocation deletion need to
+    #be in different finally blocks.
+    finally {
+        unless ($_) {
```

Comment:
I think the test output from [this comment](https://github.com/genome/genome/pull/407/files#r25272900) suggested that `$_` won't be defined inside the `finally`.

If this needs to only commit if there's not an error, it seems like it belongs inside the `try` block itself.